datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -804,10 +804,14 @@ def event(self, event):
         :param ev event: the event to be sent.
         """
         # Enforce types of some fields, considerably facilitates handling in go bindings downstream
-        for key in event:
-            # Ensure strings have the correct type
+        for key, value in iteritems(event):
+            if not isinstance(value, (text_type, binary_type)):
+                continue
+
             try:
-                event[key] = to_string(event[key])  # type: ignore
+                event[key] = to_string(value)  # type: ignore
```

Comment:
Shouldn't this actually be `to_native_string` once #5996 gets merged?

Reply:
Indeed :-) I expect to have a merge conflict here

Reply:
Updated